subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/TaskIdentity.java

```diff
@@ -52,29 +48,11 @@
     }
 
     /**
-     * Create a task identity.
-     */
-    public static <T extends Task> TaskIdentity<T> create(String name, Class<T> type, ProjectInternal project) {
-        return doCreate(
-            name,
-            type,
-            project,
-            SEQUENCE.getAndIncrement()
-        );
-    }
-
-    /**
-     * Create a task identity.
+     * Creates a task identity with a specific unique id.
      * <p>
-     * Should only be used when loading from the configuration cache to preserve task ids.
+     * This method should not be called directly, use {@link TaskIdentityFactory} instead.
      */
     public static <T extends Task> TaskIdentity<T> create(String name, Class<T> type, ProjectInternal project, long uniqueId) {
```

Comment:
This method is used in multiple tests as a helper. These tests are in different packages, so we can't make the `create` method or a corresponding constructor package-private. Rather, we can, and it seems Groovy magic would allow the test code to access a package-private method from another package. Though, you get a warning in IDE for each such usage. We also can't mock the `TaskIdentity` in the tests, because it is `final` class.

So, I technically can do what you suggest. Do you think it's fine to have `Access to 'create' exceeds its access rights` kind of IDE warnings in all the test code from different packages?

<img width="904" alt="image" src="https://user-images.githubusercontent.com/2759152/227182134-a9b7f045-3b1d-45d1-88ae-7c7381b09a3b.png">

Reply:
I'd prefer not to have those warnings. Could the tests use the factory instead?

Reply:
Addressed in another comment